RTN-023.tex

```diff
@@ -50,6 +50,372 @@
+
+The purpose of this document is to describe requirements on a campaign
+tracking and monitoring system that can begin by assisting in DP0 
+\cite{RTN-001} production and grow to assisting in DRP and other 
+data processing for Rubin.
+
+\section{Campaign Generation and Tracking Requirements}
+
+\begin{figure}
+\includegraphics[width=\textwidth]{CampaignTooling.jpg}]
+\caption{Diagram of Campaign generation (left) and monitoring system (right).
+Credit: C. Slater}
+\label{CampaignTooling}
+\end{figure}
+
+To generate processing campaigns and track them these features are desired,
+as shown in Figure \ref{CampaignTooling}:
+
+	\begin{enumerate}
+
+	\item Ability to access up-to-date source of inputs to-be-processed
+		into outputs and their current state 
+		(already processed, processed and flagged bad, 
+		in processing)
+
+	\begin{itemize}
+	
+	\item May wish to avoid constantly hitting the butler for
+	its contents -- so may wish to have another active source
+	of inputs and status of processing -- i.e. another database
+	which would only occasionally check with the butler to sync up
+
+
+	\end{itemize}
+
+	\item Ability to generate BPS submit yaml configs 
+	(or other top level workflow submit scripts) which request processing
+	of subset of inputs needing to be processed, pipelines to be run,
+	outputs described.  This could, for instance, be achieved by having
+	a set of template yaml configs -- one for each procesing step
+	(Single Frame, Consolidate Visit, Photo calib FGCM, 
+	astrometric calib skyMap, Coadd/forced Photometry, Difference
+	Imaging Analysis) where a 'sed' script or similiar would fill in
+	a visit list, a src catalog list, a tract list, etc
+
+	\begin{itemize}
+
+	\item Should be able to 'split' a large set of inputs into
+	serveral managable sized pieces.  For instance, DP0.2 consists of
+	about 40,000 visits, and a typical Rubin night of observing will
+	take 1,000-2,000 science exposures.   So a managable unit for
+	single visit processing might be of size 1000 exposures/visits and
+	a DP0.2 processing would split the 40,000 visits into 1000 visit
+	groups.  Consideration should be given to how long (wallclock time)
+	a single campaign 'unit' may be run and to not generally exceed 
+	24 hours to a few days to avoid interruptions from downtimes.
+	Also depends on ability of system to recover from interruptions and
+	continue without needing to restart the whole campaign from
+	the beginning.
+	
+
+	\item  Should allow(?) 'last minute editing' of configs before
+	submission to tweak something, change a pipeline version or
+	visit input list
+
+	\item each campaign should be identified by some unique id string, 
+	perhaps same as workflow manager identifying string?
+
+	\end{itemize}
+
+	\item Ability to store and manage a set of BPS submit scripts 
+	(or equivalent).
+	For instance, as each BPS submit yaml is performed, the yaml
```

Comment:
If we are using templating we don't necessarily need to be storing the submit yaml itself forever, only the parameters -- we can tie the submissions to the products if the templating system is what creates the output RUN collection name.